src/common/WeightedPriorityQueue.h

```diff
@@ -84,7 +84,8 @@ class WeightedPriorityQueue :  public OpQueue <T, K>
       }
       //Get the cost of the next item to dequeue
       unsigned get_cost() const {
-	return lp.begin()->cost;
+        assert(!empty());
```

Comment:
This is the kind of assert which can be compiled out, and it's pretty minimal in terms of cost.  I don't feel any need to avoid this kind of assert.  Are you worried about the overhead?